src/thread_priority.cpp

```diff
@@ -53,4 +55,49 @@ bool configure_sched_fifo(int priority)
   return !sched_setscheduler(0, SCHED_FIFO, &schedp);
 }
 
+bool is_capable(cap_value_t v)
+{
+  bool rc = false;
+  cap_t caps;
+  if ((caps = cap_get_proc()) == NULL) {
+    return false;
+  }
+
+  if (cap_set_flag(caps, CAP_EFFECTIVE, 1, &v, CAP_SET) == -1) {
+    goto cleanup;
```

Comment:
Instead of a `goto`: Could we use a lambda function there?

Reply:
Done